### PR TITLE
Very basic item hints for MSR

### PR DESCRIPTION
**Pull request by ThanatosGit**
Redone the rdvgame files in the tests because they didn't had hints before.

### CODE REVIEW

**Comment by dyceron** [comment]
Is this PR going to enable UI to change the hint precision, or is that for another pr?

**Review by dyceron — CHANGES_REQUESTED** [review]
Add tests

**Comment by ThanatosGit** [comment]
> Is this PR going to enable UI to change the hint precision, or is that for another pr?

Nope. It's really very simple. We might want to change it in the future. That depends on whether we think the clues are too powerful or too weak, as they are now.

Removed all the unneccessary functions. The generator part (`HintDistributor`) is not tested really for specific games if I'm not mistaken.

File: randovania/games/samus_returns/exporter/hint_namer.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        msg = self.location_formatters[hint.precision.location].format(
+            game,
+            dataclasses.replace(pick_hint, pickup_name=pick_hint.pickup_name),
+            hint,
+            with_color,
+        )
+        return f"{msg}\n"
+
+    def format_temple_name(self, temple_name: str, with_color: bool) -> str:
+        raise RuntimeError("Unsupported feature")
```
Comment (dyceron) [inline randovania/games/samus_returns/exporter/hint_namer.py:62]: Since color isn't a thing, shouldn't this be removed?

Comment (ThanatosGit) [reply randovania/games/samus_returns/exporter/hint_namer.py:47]: In general, yes, but in this case it's implementing the methods from the base class: https://github.com/randovania/randovania/blob/main/randovania/exporter/hints/hint_namer.py

**Review by dyceron — APPROVED** [review]

**Comment by dyceron** [comment]
We probably want to have a guaranteed hint for the baby.

**Comment by ThanatosGit** [comment]
> We probably want to have a guaranteed hint for the baby

At a specific spot or just guarantee that the hint is somewhere?

**Comment by dyceron** [comment]
> At a specific spot or just guarantee that the hint is somewhere?

We could have a specific spot (could do the same for ice later on) or for now just have it guaranteed period anywhere)

**Pull request merged**